pkg/stub/handler.go

```diff
@@ -1005,7 +1101,10 @@ func (h *Handler) Handle(ctx context.Context, event sdk.Event) error {
 		// if trying to do rhel to the default registry.redhat.io registry requires the secret
 		// be in place since registry.redhat.io requires auth to pull; if it is not ready
 		// error state will be logged by WaitingForCredential
-		err = h.WaitingForCredential(srcfg)
+		returnNil, err := h.WaitingForCredential(srcfg)
+		if returnNil {
+			return nil
+		}
```

Comment:
then make this logic:

```
waiting,err:=h.WaitingForCredential(srcfg)
if waiting && err!=nil {
      // if err!=nil, then srccfg was updated with an error condition, so we need to call sdk update
      logrus.Println("samplesresource update ignored since need the RHEL credential")
      return h.sdkwrapper.Update(srcfg)
}
if waiting {
  // we're waiting, but srcfg has not changed, so no-op
  return nil
}
// process
```

Reply:
it seems easier to reason about than "returnNil"

Reply:
you could even make waitingForCredential just return bool,bool  (waiting,srcfgModified) since you are not actually using the err object that's being returned other than checking for nil on it.